tests/functional/sources/test_source_configs.py

```diff
@@ -0,0 +1,594 @@
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": disabled_source_table__schema_yml,
+        }
+
+    @pytest.mark.xfail
+    def test_source_config_yaml_source_table(self, project):
+        run_dbt(["compile"])
+        manifest = get_manifest(project.project_root)
+        assert "source.test.test_source.test_table" in manifest.sources
+        assert "source.test.test_source.disabled_test_table" not in manifest.sources
+
+
+# Test all configs other than enabled in dbt_project.yml
+# expect fail - not implemented
+class TestAllConfigsProjectLevel(SourceConfigTests):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": models__schema_yml}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "sources": {
+                "enabled": True,
+                "quoting": {
+                    "database": False,
+                    "schema": False,
+                    "identifier": False,
+                    "column": False,
+                },
+                "freshness": {
+                    "error_after": {"count": 24, "period": "hour"},
+                    "warn_after": {"count": 12, "period": "hour"},
+                },
+                "loader": "a_loader",
+                "loaded_at_field": "some_column",
+                "database": "custom_database",
+                "schema": "custom_schema",
+                # identifier: "seed"  #this doesnt seems to be supported now?
+                "meta": {"languages": ["python"]},
+                "tags": ["important_tag"],
+            }
+        }
+
+    @pytest.mark.xfail
+    def test_source_all_configs_dbt_project(self, project):
+        run_dbt(["compile"])
+        manifest = get_manifest(project.project_root)
+        assert "source.test.test_source.test_table" in manifest.sources
+        config = manifest.sources.get("source.test.test_source.test_table").config
+
+        assert isinstance(config, SourceConfig)
+
+        assert config == pytest.expected_config
+
+
+configs_source_level__schema_yml = """version: 2
+
+sources:
+  - name: test_source
+    config:
+        enabled: True,
+        quoting:
+            database: False
+            schema: False
+            identifier: False
+            column: False
+        freshness:
+            error_after: {count: 24, period: hour}
+            warn_after: {count: 12, period: hour}
+        loader: "a_loader"
+        loaded_at_field: some_column
+        database: custom_database
+        schema: custom_schema
+        # identifier: "seed"  #this doesnt seems to be supported now?
+        meta: {'languages': ['python']}
+        tags: ["important_tag"]
+    tables:
+      - name: test_table
+      - name: other_test_table
+"""
+
+
+# Test configs other than enabled at sources level in yaml file
+# **currently passes since enabled is all that ends up in the
```

Comment:
Fair question: should we expect this to fail?

We let users set whichever configs they want in `dbt_project.yml`, regardless of whether they take any effect, since model/seed/snapshot/test configs might be used at runtime in the associated materialization. That does make it harder to raise error in the case of typos (https://github.com/dbt-labs/dbt-core/issues/2606, https://github.com/dbt-labs/dbt-core/issues/4280).

Sources aren't materialized... the closest thing I could think is, for `dbt source freshness`, if users want to pass totally custom configs for use in a custom `collect_freshness` macro. They _could_ achieve that with `config.meta`.